Source/WebCore/platform/graphics/gstreamer/GStreamerVideoFrameConverter.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2025 Igalia S.L
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "GStreamerVideoFrameConverter.h"
+
+#if USE(GSTREAMER)
+
+#include "GStreamerCommon.h"
+#include <gst/app/gstappsink.h>
+#include <gst/app/gstappsrc.h>
+#include <gst/gl/gl.h>
+#include <wtf/NeverDestroyed.h>
+#include <wtf/Scope.h>
+
+namespace WebCore {
+
+GST_DEBUG_CATEGORY_STATIC(webkit_gst_video_frame_converter_debug);
+#define GST_CAT_DEFAULT webkit_gst_video_frame_converter_debug
+
+GStreamerVideoFrameConverter& GStreamerVideoFrameConverter::singleton()
+{
+    static NeverDestroyed<GStreamerVideoFrameConverter> sharedInstance;
+    return sharedInstance;
+}
+
+GStreamerVideoFrameConverter::GStreamerVideoFrameConverter()
+{
+    ensureGStreamerInitialized();
+    GST_DEBUG_CATEGORY_INIT(webkit_gst_video_frame_converter_debug, "webkitvideoframeconverter", 0, "WebKit GStreamer Video Frame Converter");
+    m_pipeline = gst_element_factory_make("pipeline", "video-frame-converter");
+    m_src = makeGStreamerElement("appsrc", nullptr);
+    auto gldownload = makeGStreamerElement("gldownload", nullptr);
+    auto videoconvert = makeGStreamerElement("videoconvert", nullptr);
+    auto videoscale = makeGStreamerElement("videoscale", nullptr);
+    m_sink = makeGStreamerElement("appsink", nullptr);
+
+    g_object_set(m_sink.get(), "emit-signals", TRUE, nullptr);
+    if (webkitGstCheckVersion(1, 24, 0)) {
+        g_signal_connect(m_sink.get(), "propose-allocation", G_CALLBACK(+[](GstElement*, GstQuery* query) -> gboolean {
```

Comment:
Is this signal connection the reason why we set the emit signals to true? If it is like this, then we can move the set to inside the if.